LibreNMS/OS/Vrp.php

```diff
@@ -96,6 +97,59 @@ public function discoverEntityPhysical(): Collection
         return $inventory;
     }
 
+    public function discoverTransceivers(): Collection
+    {
+        // Get a map of ifIndex to port_id for proper association with ports
+        $ifIndexToPortId = $this->getDevice()->ports()->pluck('port_id', 'ifIndex');
+
+        // EntityPhysicalIndex to ifIndex
+        $entityToIfIndex = \SnmpQuery::walk('ENTITY-MIB::entAliasMappingIdentifier')->mapTable(function ($data, $entPhysicalIndex) {
```

Comment:
The entity-physical module should already be fetching this oid and storing ifIndex. You could save some work by using that, I do something similar in the Cisco code.  Perhaps I will add some common code to use to help with this.

Reply:
This actually exists :)

$entityToIfIndex = $this->getIfIndexEntPhysicalMap();

Reply:
Hey @murrant @PipoCanaja where i am getting wrong

Reply:
You can replace the SnmpQuery with `$entityToIfIndex = $this->getIfIndexEntPhysicalMap();` which will give you the same map without duplicating an snmpwalk.